pkgs/os-specific/linux/microcode/intel.nix

```diff
@@ -1,27 +1,26 @@
-{ stdenv, fetchurl, libarchive }:
+{ stdenv, fetchurl, libarchive, iucode-tool }:
 
 stdenv.mkDerivation rec {
   name = "microcode-intel-${version}";
-  version = "20180312";
+  version = "20180807";
 
   src = fetchurl {
-    url = "https://downloadmirror.intel.com/27591/eng/microcode-${version}.tgz";
-    sha256 = "0yg7q5blcqgq8jyjxhn9n48rxws77ylqzyn4kn10l6yzwan1yf0b";
+    url = "https://downloadmirror.intel.com/28039/eng/microcode-${version}.tgz";
+    sha256 = "0h4ygwx5brnrjz8v47aikrwhf0q3jhizxmzcii4bdjg64zffiy99";
   };
 
-  buildInputs = [ libarchive ];
+  nativeBuildInputs = [ iucode-tool libarchive ];
 
   sourceRoot = ".";
 
-  buildPhase = ''
-    gcc -O2 -Wall -o intel-microcode2ucode ${./intel-microcode2ucode.c}
-    ./intel-microcode2ucode microcode.dat
-  '';
-
   installPhase = ''
+    runHook preInstall
+
     mkdir -p $out kernel/x86/microcode
-    mv microcode.bin kernel/x86/microcode/GenuineIntel.bin
+    iucode_tool -w kernel/x86/microcode/GenuineIntel.bin intel-ucode{,-with-caveats}/
```

Comment:
What is intel-ucode-with-caveats?

Reply:
Arch Linux includes it (it's a relatively new addition to the microcode distribution) but the problem is that in order to safely apply them, various kernel patches must be in place. I don't know if they are, so I ripped it out again.